app/src/main/java/io/gnosis/safe/ui/transactions/TransactionListViewModel.kt

```diff
@@ -38,21 +33,23 @@ class TransactionListViewModel
     appDispatchers: AppDispatchers
 ) : BaseStateViewModel<TransactionsViewState>(appDispatchers) {
 
+    override fun initialState(): TransactionsViewState = TransactionsViewState(null, true)
+
     init {
         safeLaunch {
-            safeRepository.activeSafeFlow().collect { load(true) }
+            safeRepository.activeSafeFlow().collect { safe ->
+                updateState { TransactionsViewState(isLoading = true, viewAction = ActiveSafeChanged(safe)) }
+            }
         }
     }
 
-    override fun initialState(): TransactionsViewState = TransactionsViewState(null, true)
-
-    fun load(safeChange: Boolean = false) {
+    fun load(type: TransactionPagingSource.Type, safeChange: Boolean = false) {
         safeLaunch {
             val safe = safeRepository.getActiveSafe()
             updateState { TransactionsViewState(isLoading = true, viewAction = if (safeChange) ActiveSafeChanged(safe) else ViewAction.None) }
             if (safe != null) {
                 val owner = ownerCredentialsRepository.retrieveCredentials()?.address
```

Comment:
we don't have to check owner for history tx's

Reply:
IMHO the performance gain (one call of .contains() in a very short list) is not worth the code duplication. Let me know if you disagree, then I'll add it to the next PR.